CI fails: 403 when retrieving repo brownie-mix/github-actions 'API rate limit exceeded'

Note: This is just like [ocean.py#1217](https://github.com/oceanprotocol/ocean.py/issues/1217), but for df-py.

### The error

I just observed the following, on code that should have passed. (I'd just merged a PR, in the branch the tests passed.) [Log](https://github.com/oceanprotocol/df-py/actions/runs/3935573102/jobs/6731378481).

```
Run brownie pm install OpenZeppelin/openzeppelin-contracts@4.2.0
Brownie v1.1[9](https://github.com/oceanprotocol/df-py/actions/runs/3935573102/jobs/6731378481#step:7:10).2 - Python development framework for Ethereum
  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/pm.py", line 39, in main
    fn(*args["<arguments>"])
  File "brownie/_cli/pm.py", line 93, in _install
    package_id = project.main.install_package(uri)
  File "brownie/project/main.py", line 808, in install_package
    return _install_from_github(package_id)
  File "brownie/project/main.py", line 871, in _install_from_github
    download_url = _get_download_url_from_tag(org, repo, version, headers)
  File "brownie/project/main.py", line 947, in _get_download_url_from_tag
    raise ConnectionError(msg)
ConnectionError: Status 403 when getting package versions from Github: 'API rate limit exceeded for 172.177.209.209. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)'

Missing or forbidden.
If this issue persists, generate a Github API token and store it as the environment variable `GITHUB_TOKEN`:
https://github.blog/20[13](https://github.com/oceanprotocol/df-py/actions/runs/3935573102/jobs/6731378481#step:7:14)-05-[16](https://github.com/oceanprotocol/df-py/actions/runs/3935573102/jobs/6731378481#step:7:17)-personal-api-tokens/

Error: Process completed with exit code 1.
```

### TODO

Do like [ocean.py#1217](https://github.com/oceanprotocol/ocean.py/issues/1217), but for df-py, in test.yml

Look at [ocean.py's pytest.yml](https://github.com/oceanprotocol/ocean.py/blob/main/.github/workflows/pytest.yml), with the lines on GITHUB_TOKEN.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -7,11 +7,12 @@ on:
     branches: [main]
   workflow_call:
 
+env:
+  GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
 jobs:
   build:
     runs-on: ubuntu-latest
-    env:
-      GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
     strategy:
       matrix:
         python-version: [3.8]

